qa/tasks/cephadm.py

```diff
@@ -1606,7 +1606,7 @@ def task(ctx, config):
 
     if not hasattr(ctx.ceph[cluster_name], 'image'):
         ctx.ceph[cluster_name].image = config.get('image')
-    ref = None
+    ref = ctx.config.get("branch", "main")
```

Comment:
Would it make more sense to update this line to `ref = ""` and update line 1628 below?

Reply:
I don't really think so, as IIRC my runs were not hitting the conditional at 1612 below. The task would fail due to a `TypeError` [here](https://github.com/zmc/ceph/blob/osd-containers-tasks/qa/tasks/cephadm.py#L171).

Reply:
OK. So I think that the only case where that can happen is if `ctx.ceph[cluster_name].image` is already set. The if-block immediately following the current `ref = None` appears to be trying to do two things - set the image & (as a side effect) set ref. 

Now, my suggestion is to keep your change but also simplify the else branch below:
```python
        else:
            # hmm, fall back to branch?
            ctx.ceph[cluster_name].image = container_image_name + ':' + ref
```

Reply:
done - good suggestion!